blocks_vertical/procedures.js

```diff
@@ -233,10 +244,15 @@ Blockly.ScratchBlocks.ProcedureUtils.deleteShadows_ = function(connectionMap) {
   // Get rid of all of the old shadow blocks if they aren't connected.
   if (connectionMap) {
     for (var id in connectionMap) {
-      var block = connectionMap[id];
-      if (block && block.isShadow()) {
-        block.dispose();
-        connectionMap[id] = null;
+      var saveInfo = connectionMap[id];
+      if (saveInfo) {
+        var block = saveInfo['block'];
+        if (block && block.isShadow()) {
+          block.dispose();
+          connectionMap[id] = null;
+          // At this point we know which shadow DOMs are about to be orphaned in
+          // the VM.  What do we do with that information?
```

Comment:
These get delete events right (thanks to dispose)?

Reply:
Shadow blocks get delete events.  But an occluded shadow is not a block that's being disposed, and does not get a delete event.

Reply:
Hmmm so my thought in response to this comment is:
- it would be nice to somehow provide info to the VM that the shadow that got deleted was because of a shadow getting obscured (as opposed to the input containing the shadow getting deleted from the block)

That way, the VM can continue to 'ignore' delete events for shadows deleted because they were obscured, and in the case that a shadow was deleted because the input that contains that shadow is deleted from the parent block, we can actually handle that shadow deletion in the vm directly (let me know if this doesn't make any sense)

At any rate, the above is only a nice to have, and if it requires a ton of extra work, it does not need to get done. This logic can feasibly be checked in the VM itself, but I seem to have found a different bug that needs to be fixed before that can happen =/

Reply:
I'll merge this and we can circle back if needed.  My first thought is that we can fire events when changing the shadow dom, with the text of the shadow dom as the event info, and that should probably be enough.